File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> solution;
        for(int i = 0; i<nums.size(); i++){
            for(int j = 0; j<nums.size(); j++){
                if(nums[i] + nums[j] == target && i != j){
                    solution.push_back(min(i,j));
                    solution.push_back(max(i,j));
                    return solution;
                }
            }
        }
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        self.res = []
        self.target = target
        self.length = len(nums)
        def search(arr, s, index):
            if s == self.target:
                self.res.append(arr)
                return
            elif s > self.target:
                return
            for i in range(index, self.length):
                arr.append(nums[i])
                search(arr.copy(), s + nums[i], i)
                arr.pop()
        search([], 0, 0)
        return self.res
